lib/ChildUtils.spec.lua

```diff
@@ -0,0 +1,84 @@
+return function()
+	local ChildUtils = require(script.Parent.ChildUtils)
+	local createElement = require(script.Parent.createElement)
+	local Type = require(script.Parent.Type)
+
+	describe("iterateChildren", function()
+		it("should iterate once for a single child", function()
+			local child = createElement("TextLabel")
+			local iterator = ChildUtils.iterateChildren(child)
+			local iteratedKey, iteratedChild = iterator()
+			-- For single elements, the key should be UseParentKey
+			expect(iteratedKey).to.equal(ChildUtils.UseParentKey)
+			expect(iteratedChild).to.equal(child)
+
+			iteratedKey = iterator()
+			expect(iteratedKey).to.equal(nil)
+		end)
+
+		it("should iterate over multiple children", function()
+			local children = {
+				a = createElement("TextLabel"),
+				b = createElement("TextLabel"),
+			}
+
+			local count = 0
+
+			for key, child in ChildUtils.iterateChildren(children) do
+				expect(typeof(key)).to.equal("string")
+				expect(Type.of(child)).to.equal(Type.Element)
```

Comment:
We should check that we're getting the right children, too!

Reply:
Done! It's kind of troublesome because of no guarantee around iteration order x.x